src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -91,4 +101,31 @@ public Command parseCommand(String userInput) throws ParseException {
         }
     }
 
+    private boolean havePendingCommand() {
+        return this.pendingCommand != null;
+    }
+
+    /**
+     * Parses user input into command for executing pending command.
+     *
+     * @param userInput full user input string
+     * @return the command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public Command executePendingCommand(String userInput) throws ParseException {
+
+        if (userInput.equalsIgnoreCase("Y")) {
```

Comment:
Should execute pending command be just execute whatever command is holding rather than this specific "Y" vs "N"?